subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppApplicationPlugin.java

```diff
@@ -101,8 +102,8 @@ public void execute(final Project project) {
 
                 for (BuildType buildType : BuildType.DEFAULT_BUILD_TYPES) {
                     for (TargetMachine targetMachine : targetMachines) {
-                        String operatingSystemSuffix = createDimensionSuffix(targetMachine.getOperatingSystemFamily(), targetMachines);
-                        String architectureSuffix = createDimensionSuffix(targetMachine.getArchitecture(), targetMachines);
+                        String operatingSystemSuffix = createDimensionSuffix(targetMachine.getOperatingSystemFamily(), targetMachines.stream().map(TargetMachine::getOperatingSystemFamily).collect(Collectors.toSet()));
```

Comment:
We should get rid of all of this duplication. There should be one method/class somewhere that makes the decision about what names are used for a variant, rather than reimplementing this in each plugin.

Reply:
I agree with this. I would say this is out-of-scope of this PR. I tried something out but didn't simplify the code except move the common code in the same place and had issues that would have slowed down the review of this PR. I would suggest we deal with that specific issue separately so we don't delay this PR.

Reply:
Issue https://github.com/gradle/gradle-native/issues/954 will track this debt.